Filesystem.cpp

```diff
@@ -8,49 +8,62 @@ root=new Directory("root",nullptr);
 currentDir=root;
 }
 FileSystem::~FileSystem(){
-for(auto& i : Directorys){
+for(auto& i : listOfFolders){
     delete i;
 }
 delete root;
 }
+bool FileSystem::hasDir(string arg){
+    for(auto i:currentDir->getSubFolders()){
+        if(i->getName()==arg)return true;
+    }
+    return false;
+}
+bool FileSystem::hasFile(string arg){
+    for(auto i:currentDir->getFiles()){
+        if(i->getName()==arg)return true;
+    }
+    return false;
+}
 
 
 void FileSystem::touch(vector <string> args){
-    if(currentDir->noFile()){
-        //push a new file
-        currentDir->makeFile(args.front());
-        return;
-    }
-        //check that the file is already exits
-    for(auto i:currentDir->getFile()){
-        if(i->getName()==args.front()){
-            cout<<"The file is already exits"<<endl;
+    bool hasD = hasDir(args.front());
+    bool hasF = hasFile(args.front());
+    if(hasD || hasF){
+        if(hasD){
+            cout<<"There is already a folder with this name!"<<endl;
+            return;
+        }
+        if(hasF){
+            cout<<"The file is already exits!"<<endl;
             return;
         }
     }
```

Comment:
namost pepitaban ugyanennek a logikanak meg kellene lennie pl az mkdir-ben is, nem? (de ott mas van). Ha pedig mar ket helyen meg kellene lennie....